Panic in the AttachDetachController

**What happened**:
During investigation of  nil pointer dereference panic, i've found that [operationGenerator.GenerateDetachVolumeFunc](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_generator.go#L389) has an issue:

`operationGenerator.GenerateDetachVolumeFunc` don't check 1st return value of `volumePluginMgr.FindAttachablePluginByName`  to `nil`: 

```
                attachableVolumePlugin, err = og.volumePluginMgr.FindAttachablePluginByName(pluginName)
		if err != nil {
			return volumetypes.GeneratedOperations{}, volumeToDetach.GenerateErrorDetailed("DetachVolume.FindAttachablePluginBySpec failed", err)
		}
	}

	if pluginName == "" {
		pluginName = attachableVolumePlugin.GetPluginName()
	}

	volumeDetacher, err := attachableVolumePlugin.NewDetacher()
```
but [VolumePluginMgr.FindAttachablePluginByName](https://github.com/kubernetes/kubernetes/blob/fad23990cea2f883c98684fc2f2b600a3c1bdbd8/pkg/volume/plugins.go#L800) could return  `nil, nil`:

```
func (pm *VolumePluginMgr) FindAttachablePluginByName(name string) (AttachableVolumePlugin, error) {
	volumePlugin, err := pm.FindPluginByName(name)
	if err != nil {
		return nil, err
	}
	if attachablePlugin, ok := volumePlugin.(AttachableVolumePlugin); ok {
		return attachablePlugin, nil
	}
	return nil, nil
}
```

So, if `VolumePluginMgr.FindAttachablePluginByName()` finds plugin by name, but this plugin isn't a `AttachableVolumePlugin`, it will cause a nil pointer dereference panic in `volumeDetacher, err := attachableVolumePlugin.NewDetacher()`

**What you expected to happen**:
`operationGenerator.GenerateDetachVolumeFunc` should check `attachableVolumePlugin == nil`

**How to reproduce it (as minimally and precisely as possible)**:
Unfortunately, i can't reproduce it

**Anything else we need to know?**:
@liggitt advised to change `FindAttachablePluginByName` signature to `(plugin, bool, error)`

<!-- DO NOT EDIT BELOW THIS LINE -->
/kind bug

@gnufied I've sent a PR to fix this issue

@saad-ali 
We faced it in about 30 minutes during rolling update, and, unfortunately, we're unable to reproduce it.
It was k8s with modified apiserver & kubelet, but there was no modifications in kube-controller-manager.

Panic trace:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x2f0f024]

goroutine 1502 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x108
panic(0x3dc8dc0, 0xa10c620)
/usr/local/go/src/runtime/panic.go:513 +0x1b9
k8s.io/kubernetes/pkg/volume/util/operationexecutor.(*operationGenerator).GenerateDetachVolumeFunc(0xc001e28a80, 0xc00dc559a0, 0x1e, 0x0, 0xc00dc55720, 0x14, 0x1, 0x0, 0x0, 0x1, ...)
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/volume/util/operationexecutor/operation_generator.go:367 +0x1b4
k8s.io/kubernetes/pkg/volume/util/operationexecutor.(*operationExecutor).DetachVolume(0xc001e63520, 0xc00dc559a0, 0x1e, 0x0, 0xc00dc55720, 0x14, 0x1, 0x0, 0x0, 0x431601, ...)
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/volume/util/operationexecutor/operation_executor.go:590 +0xa3
k8s.io/kubernetes/pkg/controller/volume/attachdetach/reconciler.(*reconciler).reconcile(0xc001b87560)
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/volume/attachdetach/reconciler/reconciler.go:224 +0x659
k8s.io/kubernetes/pkg/controller/volume/attachdetach/reconciler.(*reconciler).reconciliationLoopFunc.func1()
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/volume/attachdetach/reconciler/reconciler.go:107 +0x33
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0098b9590)
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0098b9590, 0x5f5e100, 0x0, 0x3f81c01, 0xc0002230e0)
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbe
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0098b9590, 0x5f5e100, 0xc0002230e0)
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
k8s.io/kubernetes/pkg/controller/volume/attachdetach/reconciler.(*reconciler).Run(0xc001b87560, 0xc0002230e0)
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/volume/attachdetach/reconciler/reconciler.go:98 +0x50
created by k8s.io/kubernetes/pkg/controller/volume/attachdetach.(*attachDetachController).Run
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/volume/attachdetach/attach_detach_controller.go:322 +0x2a1
```

--- a/pkg/volume/util/operationexecutor/operation_generator.go
+++ b/pkg/volume/util/operationexecutor/operation_generator.go
@@ -387,8 +387,8 @@ func (og *operationGenerator) GenerateDetachVolumeFunc(
 			return volumetypes.GeneratedOperations{}, volumeToDetach.GenerateErrorDetailed("DetachVolume.SplitUniqueName failed", err)
 		}
 		attachableVolumePlugin, err = og.volumePluginMgr.FindAttachablePluginByName(pluginName)
-		if err != nil {
-			return volumetypes.GeneratedOperations{}, volumeToDetach.GenerateErrorDetailed("DetachVolume.FindAttachablePluginBySpec failed", err)
+		if err != nil || attachableVolumePlugin == nil {
+			return volumetypes.GeneratedOperations{}, volumeToDetach.GenerateErrorDetailed("DetachVolume.FindAttachablePluginByName failed", err)
 		}
 	}
 

